File: jmlparser-jml-tools/src/test/java/com/github/jmlparser/lint/LinterTest.java
package com.github.jmlparser.lint;

import com.github.jmlparser.TestWithJavaParser;
import org.junit.jupiter.api.Assumptions;
import org.junit.jupiter.api.Test;

import java.util.Collections;

/**
 * @author Alexander Weigl
 * @version 1 (14.10.22)
 */
class LinterTest extends TestWithJavaParser {
    @Test
    void everythingWrong() {
        var result = parser.parse(getClass().getResourceAsStream("EverythingWrong.java"));
        Assumptions.assumeTrue(result.isSuccessful());
        var actual =
                new JmlLintingFacade(new JmlLintingConfig()).lint(Collections.singletonList(result.getResult().get()));

        for (LintProblem lintProblem : actual) {
            System.out.println(lintProblem);
        }
    }
}

File: tools/lint/src/test/kotlin/LinterTest.kt
import io.github.jmltoolkit.lint.JmlLintingConfig
import io.github.jmltoolkit.lint.JmlLintingFacade
import io.github.jmltoolkit.utils.TestWithJavaParser
import org.junit.jupiter.api.Assumptions
import org.junit.jupiter.api.Test

/**
 * @author Alexander Weigl
 * @version 1 (14.10.22)
 */
internal class LinterTest : TestWithJavaParser() {
    @Test
    fun everythingWrong() {
        val result = parser.parse(javaClass.getResourceAsStream("EverythingWrong.java"))
        Assumptions.assumeTrue(result.isSuccessful)
        val actual = JmlLintingFacade(JmlLintingConfig()).lint(listOf(result.result.get()))

        for (lintProblem in actual) {
            println(lintProblem)
        }
    }
}
